### PR TITLE
Branch filter event command

### PR DESCRIPTION
**Pull request by XXJJXJ**
This PR adds the `filterEvents` feature and makes some changes to a number of files. 

Things to note:
1. Model::getEventList() method have been renamed to Model::getFilteredEventList() to accurately describe what this method actually does.
2. Event class and its components have been changed/improved, to fix a bug and allow setting an Event at 0000hr. Most changes occur in EventTime class.
3. Additional feature `listEvents` added to recover unfiltered list of events.
4. TestFiles are included with minimal testcases and will need some work when more features in v1.2 are implemented.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FilterEventCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import seedu.address.model.event.EventDate;
+import seedu.address.model.event.EventDateTimePredicate;
+import seedu.address.model.event.EventTime;
+
+public class FilterEventCommandParser implements Parser<FilterEventCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FilterEventCommand
+     * and returns a FilterCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
```
Comment (lowkangn) [inline src/main/java/seedu/address/logic/parser/FilterEventCommandParser.java:21]:
```suggestion
     * @throws ParseException if the user input does not conform to the expected format
```

File: src/test/java/seedu/address/logic/commands/ListEventCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for ListEventCommand.
+ */
+public class ListEventCommandTest {
+
+    private Model model;
+    private Model expectedModel;
+
+    @BeforeEach
+    public void setUo() {
```
Comment (lowkangn) [inline src/test/java/seedu/address/logic/commands/ListEventCommandTest.java:25]: Possible typo

**Review by lowkangn — APPROVED** [review]
LGTM, just a few typos that I have pointed out

File: src/test/java/seedu/address/logic/commands/ListEventCommandTest.java (hunk 2 above)
Comment (XXJJXJ) [reply src/test/java/seedu/address/logic/commands/ListEventCommandTest.java]: Yeap it is lol. Fixed it.

**Pull request merged**